File: 2018/puzzle1.py
from collections import defaultdict


def read_data():
    with open('puzzle1.dat') as f:
        return [int(line.strip()) for line in f]

data = read_data()

print ('Part 1:', sum(data))

occurrences = defaultdict(int)
index = 0
running_total = 0
while True:
    running_total += data[index]
    occurrences[running_total] += 1
    if occurrences[running_total] == 2:
        print ('Part 2:', running_total)
        break
    index = (index + 1) % len(data)

File: 2022/com/chasingdings/adventofcode/y2022/Puzzle1.java
package com.chasingdings.adventofcode.y2022;

import java.io.IOException;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class Puzzle1 extends AbstractPuzzle {
    private static final String DATA_FILE = "2022\\puzzle1.dat";

    @Override
    public void solve() throws IOException {
        var dataList = getInputData();
        System.out.println(String.format("Part 1: %d", dataList.get(0)));
        var total = IntStream.range(0, 3).map(i -> dataList.get(i)).sum();
        System.out.println(String.format("Part 2: %d", total));
    }

    @Override
    public String getPuzzleName() {
        return "Day 1 - Calorie Counting";
    }

    private List<Integer> getInputData() throws IOException {
        var content = readData(DATA_FILE);

        return Arrays.stream(content.split(EOL + EOL))
                .map(elf -> Arrays.stream(elf.split(EOL)).mapToInt(Integer::parseInt).sum())
                .sorted(Collections.reverseOrder())
                .collect(Collectors.toList());
    }
}
